### DISCUSSION TITLE
TopoLVM with SAN attached storage

### DISCUSSION THREAD

**Discussion by romdalf** [topolvm/topolvm#964] [Q&A] [answered]
A few questions regarding a bare metal cluster with multipathed SAN devices attached to all the nodes, the LVM would then be available for the entire set of nodes:

- When draining or deleting a node, will this impact the entire LVM configuration?
- Is the topology key used to define the node constraint an immutable value?
- Do you have any configuration advice regarding such setups?

Thank you!
Rom

**Comment by satoru-takeuchi** [comment]
I'll answer in several days.

**Comment by satoru-takeuchi** [comment, marked as answer]
@romdalf

> A few questions regarding a bare metal cluster with multipathed SAN devices attached to all the nodes, the LVM would then be available for the entire set of nodes:
> - When draining or deleting a node, will this impact the entire LVM configuration?

I guess yes. However, I'm not sure what will happen in detail because both topolvm and lvm don't take such the configuration into account. Both tools assume to be used from one node.

> - Is the topology key used to define the node constraint an immutable value?

Yes. It's passed from `external-provisioner` sidecar container. (ref. [the design document](https://github.com/topolvm/topolvm/blob/main/docs/design.md#how-dynamic-provisioning-works)) 

> - Do you have any configuration advice regarding such setups?

Unfortunately, I don't have any good idea.
